### ISSUE TITLE
Document that <extension>true<extension> is only needed for custom packagings (docker. docker-tar etc)

### ISSUE THREAD

**Issue by jakub-bochenski** [fabric8io/docker-maven-plugin#540]
I found out the above by browsing the source code -- the documentation says the `<extensions>` element should always be set to `true`

Please note enabling extensions makes Maven use  multiple ClassLoaderContexts and triggers those bugs in core plugins: 
https://issues.apache.org/jira/browse/MINSTALL-102
https://issues.apache.org/jira/browse/MDEPLOY-193

Not using the plugin extensions is a nice workaround for above bugs if one doesn't use the custom packagings.

**Comment by rhuss** [comment]
Yes, the `<extensions>true</extensions>` is really for custom lifecycles.

Thanks for the heads up, I will update the documentation accordingly.

**Comment by rhuss** [comment]
Hmm, could found the spot where the docs says the `<extensions>true</extensions>` is always required. Do you remember where you found this ?

The only spot in the documentation refering to `<extensions>` is:

> This will create the jar (if any), build the Docker images, start the configured Docker containers, runs the integration tests, stops the configured Docker container when you enter `mvn install`. With `mvn deploy` you can additionally push the images to a Docker configuration. Please note the `<extensions>true</extensions>` which is mandatory when you use a custom lifecycle.

**Comment by jakub-bochenski** [comment]
Huh? I could swear the wording there was different, but git history is there to prove me wrong :)

Optionally maybe you could emphasize it more or put this also into FAQ for blind people like me.

**Comment by rhuss** [comment]
;-) Ok, will emphasize it a bit more (with links to the issues you mentioned). Thanks for the heads up!

**Comment by jakub-bochenski** [comment]
Thanks -- one more suggestion: I think maybe I got confused re. recent changes to execute `initialize` before docker goals (#505)

 Maybe changing
>  which is mandatory when you use a custom lifecycle.
to
>  which is mandatory when you use a custom packaging
would be clearer.

**Comment by rhuss** [comment]
Good idea. will add that.

**Comment by jakub-bochenski** [comment]
A workaround has been published here in that let's you use both extensions and `deployAtEnd`: https://issues.apache.org/jira/browse/MDEPLOY-193?focusedCommentId=16207644&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16207644
